src/gluonts/dataset/pandas.py

```diff
@@ -222,10 +222,18 @@ def pair_with_item_id(obj: Union[Tuple, pd.DataFrame, pd.Series]):
     raise ValueError("input must be a pair, or a pandas Series or DataFrame.")
 
 
-def infer_freq(index: pd.Index):
+def infer_freq(index: pd.Index) -> str:
     if isinstance(index, pd.PeriodIndex):
         return index.freqstr
-    return pd.infer_freq(index)
+
+    freq = pd.infer_freq(index)
+    # pandas likes to infer the `start of x` frequency, however when doing
+    # df.to_period("<x>S"), it fails, so we avoid using it. It's enough to
+    # remove the trailing S, e.g `MS` -> `M
+    if len(freq) > 1 and freq.endswith("S"):
+        return freq[:-1]
+
+    return freq
 
```

Comment:
Reading in about [offset-aliases](https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#offset-aliases), it seems that `MS` and `M` have different connotations, meaning:

- M : month end frequency
- MS : month start frequency

Do we know if we want to keep this differences or just oversee them? 

Nevertheless, both of them are monthly frequency. I just wonder how critical this difference can be.

Reply:
For periods, only `M` makes sense, since there is no day information unlike for timestamps. Pandas throws an error if you try to create a `Period` with an `xS` frequency.